types/index.d.ts

```diff
@@ -0,0 +1,116 @@
+import { ObjectSchema, Schema } from 'yup'
+import { ReactElement, ReactNode, SyntheticEvent, ReactType } from 'react'
+import { RenderedAsComponent } from './helpers'
+
+type Errors = Record<string, any[]>
+type Touched = Record<string, boolean>
+
+export type SchemaValue<T extends ObjectSchema<unknown>> = ReturnType<T['cast']>
+
+type ValidateData = {
+  fields: string[]
+  type: SyntheticEvent
+  args?: any[]
+}
+
+type BeforeSubmitData<T> = {
+  value: T
+  errors: Errors
+}
+
+type Mapper<TOut, TIn = any> = (input: TIn) => TOut
```

Comment:
weird order 🤔 If you are doing it to save typing `any` I don't think that's the end of the world. but it's unintuitive that the first are is out and second arg is in